sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1008,8 +1015,21 @@ private boolean isManagementToken(String lockToken) {
 
     private Mono<Void> updateDisposition(ServiceBusReceivedMessage message, DispositionStatus dispositionStatus,
         String deadLetterReason, String deadLetterErrorDescription, Map<String, Object> propertiesToModify,
-        ServiceBusTransactionContext transactionContext) {
+        ServiceBusTransactionContext transactionContext, ServiceBusErrorSource errorSource) {
+        return updateDispositionInternal(message, dispositionStatus, deadLetterReason, deadLetterErrorDescription,
+            propertiesToModify, transactionContext)
+            .onErrorMap(throwable -> {
+                if (throwable instanceof AmqpException) {
+                    return new ServiceBusException((AmqpException) throwable, errorSource);
+                }
+                return throwable;
 
+            });
+    }
+
+    private Mono<Void> updateDispositionInternal(ServiceBusReceivedMessage message, DispositionStatus dispositionStatus,
```

Comment:
I don't understand why we created a function here. The method itself is already private.